explainer.md

```diff
@@ -1,22 +1,26 @@
-## Explainer
+# Explainer
 
-Author: Eric Willigers &lt;<ericwilligers@chromium.org>&gt;<br>
-Author: Jay Harris &lt;<harrisjay@chromium.org>&gt;<br>
-Author: Raymes Khoury &lt;<raymes@chromium.org>&gt;
+Authors: 
+* Eric Willigers &lt;<ericwilligers@chromium.org>&gt;<br>
+* Jay Harris &lt;<harrisjay@chromium.org>&gt;<br>
+* Darwin Huang &lt;<huangdarwin@chromium.org>&gt;<br>
+* Raymes Khoury &lt;<raymes@chromium.org>&gt;
 
-### Motivation
+## Motivation
 
-This proposal gives web applications a way to register their ability to handle (read, stream, edit) files with given MIME types and/or file extensions.
+This proposal gives installed web applications a way to register their ability to handle (read, stream, edit) files with given MIME types and/or file extensions. This then allows an operating system's file manager or other native UX to select a PWA to handle the file.
```

Comment:
I read the spec for launch events the other day and I noticed that installation is only a recommendation in that spec. Could be worth following suit here?

*Note:* I no longer work on this stuff

Reply:
Ooh nice to see you, and thank you for your input! 

We need a strong signal that a user "trusts" this web application. Installation seems to be that signal for now (also per Chrome's existing DevTrial impl). I'm not immediately sure how else we might expose this, so I'm inclined to specify the behavior we currently plan for and consider opening up this API for other use cases in later iterations. 

One alternative we considered was use of a permission prompt to gate this API, but we determined it could be difficult to craft a string understandable to normal users that can properly communicate the dangers of this API. 

WDYT?

Reply:
One option we brainstormed way back was that we could add a general "open with a web application" option for the drive-by web, which would cause chrome to pop up some list of all websites you've visited recently or something that could handle that file type. Or even ones you haven't visited (but that might be weird).

Reply:
Hmm a general "open with a web application" for the drive-by web does sound nice for non-installed apps.

I can omit the word "installed", so that this can be delegated by the implementation for now. From there, we can certainly make this more clear if requested. Does that sound ok?

Reply:
Oops never mind, I realized that installation as a requirement is sprinkled throughout this explainer. I think most partners currently interested in File Handling do also strongly prefer having fewer UX steps to access files via file handling, so it might be best not to allow the drive-by web access to this API for now if that might require an extra step (for Chrome to give a list of sites that can handle the type).

I do think it would be nice to someday iterate on this and provide a way for the drive-by web to access it, but it would also be much easier on our security model if this weren't possible, so perhaps we can explore this further in a later iteration?

Reply:
Yeah, honestly it seems way more sensible to only expose this to installed web applications at first. I was mostly just interested from a consistency point of view.

Here's the relevant section from the Launch Events explainer:
https://github.com/WICG/sw-launch/blob/master/explainer.md#restricting-launch-events-to-installed-websites

(it surprises me that it's only a recommendation there to be honest, it seems like it'd have many of the same problems as this, if arbitrary websites could use launch events).

Reply:
Let's continue this discussion on #47.